scripts/referrerRevenueSummary.ts

```diff
@@ -0,0 +1,156 @@
+import { stringify } from 'csv-stringify/sync'
+import { writeFileSync } from 'fs'
+import yargs from 'yargs'
+import calculateRevenueHandlers from './calculateRevenue/protocols'
+import { protocolFilters } from './protocolFilters'
+import { NetworkId, Protocol, protocols, ReferralEvent } from './types'
+import { supportedNetworkIds } from './utils/networks'
+import { fetchReferralEvents, removeDuplicates } from './utils/referrals'
+
+async function getArgs() {
+  const argv = await yargs
+    .env('')
+    .option('protocol', {
+      alias: 'p',
+      description: 'protocol that the referrals are for',
+      demandOption: true,
+      choices: protocols,
```

Comment:
We don't actually know these, right?  Like, the way we do this is my manually submitting a tx (e.g., via a bloc explorer) vs editing a typescript file and running the script.  Remove `choices` and return an empty result if the protocol isn't registered?

Reply:
Setting the protocol is required because of how the `protocolFilter` and `calculateRevenueHandlers` are used. Is the hope that we would make this optional?